src/Hedgehog/Property.fs

```diff
@@ -81,11 +81,32 @@ module Property =
     let bind (k : 'a -> Property<'b>) (m : Property<'a>) : Property<'b> =
         bindGen (toGen << k) (toGen m) |> ofGen
 
+#if !FABLE_COMPILER
+    let private (|IsGenericList|_|) (candidate : obj) : seq<obj> option =
+        let t = candidate.GetType()
+        // have to use TypeInfo due to targeting netstandard 1.6
+        let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
+        let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
+        if isList
+        then Some (candidate :?> System.Collections.IEnumerable |> Seq.cast<obj>)
```

Comment:
Is it possible to directly cast to `ResizeArray<obj>`?

Reply:
`candidate :?> ResizeArray<obj>` throws at runtime
```
Unable to cast object of type 'System.Collections.Generic.List`1[System.Int32]' to type 'System.Collections.Generic.List`1[System.Object]
```
Possibly due to F# issues with variance https://github.com/fsharp/fslang-suggestions/issues/162

Reply:
(Weird.  I thought I already posted this.  Oh well.  Here it is again.)

Oh, yes.  That makes sense.  Otherwise, it would be possible to add any instance of type `obj`, which is not possible.